cloudevents/abstract/event.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import typing
+from typing import TypeVar
+
+
+class CloudEvent(abc.ABC):
+    """
+    Used internally for multi-framework event type integration
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "CloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _attributes_read_model(self) -> typing.Dict[str, typing.Any]:
+        """
+        :return: attributes of this event
+
+        you MUST NOT mutate this dict
+        implementation MAY assume the dic will not be mutated
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _data_read_model(self) -> typing.Optional[typing.Any]:
+        """
+        :return: data value of the  event
+        you MUST NOT mutate this value
+        implementation MAY assume the value will not be mutated
+        """
+        pass
+
+    def __setitem__(self, key: str, value: typing.Any) -> None:
+        raise NotImplementedError()
+
+    def __delitem__(self, key: str) -> None:
+        raise NotImplementedError()
+
+    def __eq__(self, other: typing.Any) -> bool:
+        if isinstance(other, CloudEvent):
+            return (
+                self._data_read_model == other._data_read_model
+                and self._attributes_read_model == other._attributes_read_model
+            )
+        return False
+
+    # Data access is handled via `.data` member
+    # Attribute access is managed via Mapping type
+    def __getitem__(self, key: str) -> typing.Any:
+        return self._attributes_read_model[key]
```

Comment:
This comment should clearly be just pydocs.

Reply:
got it